### PR TITLE
keystone: Use apache frontend

### PR DESCRIPTION
**Pull request by toabctl**
The native frontend (eventlet based) is no longer available and upstream
recommends to use a "real" webserver so use Apache.

### CODE REVIEW

File: chef/data_bags/crowbar/template-keystone.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "attributes": {
     "keystone": {
       "debug": false,
-      "frontend": "native",
+      "frontend": "apache",
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/template-keystone.json]: you need to add a migration for this.

**Comment by aplanas** [comment]
Note: keystone/recipes/server.rb still try to start the service

File: chef/data_bags/crowbar/template-keystone.json (hunk 1 above)
Comment (toabctl) [inline chef/data_bags/crowbar/template-keystone.json]: @dirkmueller fixed

**Comment by toabctl** [comment]
> Note: keystone/recipes/server.rb still try to start the service

Where? It tries to stop and I think we should allow to fail there (if the init script no longer exists after an upgrade)

**Comment by aplanas** [comment]
@toabctl but this will work in a non-upgrade scenario? There is not keystone service, but there is a `:disable, :stop` action for the frontend `apache` branch.

**Comment by toabctl** [comment]
@aplanas hm. true. we should allow to fail the stop action. Or remove it completely?

**Comment by aplanas** [comment]
@toabctl not sure, I think that during the upgrade process all the services are down, so maybe is ok to remove it. @jsuchome can provide more info.

**Comment by toabctl** [comment]
@aplanas better now?

**Comment by aplanas** [comment]
something is missing in the `auth_token` configuration: apache wakeup properly but the keystone_register wakeup service fails. The logs shows 

    File "/usr/lib/python2.7/site-packages/keystonemiddleware/auth_token/__init__.py", line 542, in _fetch_token raise NotImplemented()
        TypeError: 'NotImplementedType' object is not callable

This method is implemented in `AuthProtocol` class. Keystone-paste looks OK

**Comment by aplanas** [comment]
I found the problem in the keystonemiddleware package. The current version is 4.0.0, but newton requires >=4.5.0 <=4.9.0

**Comment by toabctl** [comment]
> I found the problem in the keystonemiddleware package. The current version is 4.0.0, but newton requires >=4.5.0 <=4.9.0

Yeah. Sorry. I thought I fixed that already but the package build fails in D:C:7:Newton.

**Comment by toabctl** [comment]
@dirkmueller are the changes ok? Can we merge this?

**Comment by toabctl** [comment]
> Only one last suggestion here: update the wsgi from https://github.com/openstack/keystone/blob/stable/newton/httpd/wsgi-keystone.conf

@aplanas I think @rhafer already synced that a while ago with the config upstream provides.

**Pull request merged**